test/MultitokenCollector.test.ts

```diff
@@ -0,0 +1,409 @@
+import {
+  FakeContract,
+  MockContract,
+  MockContractFactory,
+  smock,
+} from '@defi-wonderland/smock';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { ethers } from 'hardhat';
+import { Wallet, BigNumber } from 'ethers';
+import {
+  ERC20,
+  MultitokenCollector,
+  MultitokenCollector__factory,
+} from '../typechain-types';
+
+chai.use(chaiAsPromised);
+
+const expect = chai.expect;
+
+const PARTNER_SHARES = [20, 30, 40, 10];
+
+type Partner = {
+  beneficiary: string;
+  share: number;
+};
+
+const buildPartners = async (
+  partnerWallets: SignerWithAddress[],
+  shares: number[]
+) => {
+  return Promise.all(
+    partnerWallets.map(async (wallet, index) => {
+      const partner: Partner = {
+        beneficiary: await wallet.getAddress(),
+        share: shares[index],
```

Comment:
There is a typescript error here regarding. `Type 'number | undefined' is not assignable to type 'number'.`

Reply:
resolved with noUncheckedIndexedAccess